includes/multi-currency/class-multi-currency.php

```diff
@@ -72,14 +86,24 @@ private function __construct() {
 	public function init() {
 		include_once WCPAY_ABSPATH . 'includes/multi-currency/class-currency.php';
 		include_once WCPAY_ABSPATH . 'includes/multi-currency/class-country-flags.php';
+		include_once WCPAY_ABSPATH . 'includes/multi-currency/class-frontend-prices.php';
+		include_once WCPAY_ABSPATH . 'includes/multi-currency/class-frontend-currencies.php';
 
 		$this->id = 'wcpay_multi_currency';
 		$this->get_available_currencies();
 		$this->get_default_currency();
 		$this->get_enabled_currencies();
 
-		add_action( 'init', [ $this, 'update_selected_currency_by_url' ] );
 		add_action( 'rest_api_init', [ __CLASS__, 'init_rest_api' ] );
+
+		$is_frontend_request = ( ! is_admin() || defined( 'DOING_AJAX' ) ) && ! defined( 'DOING_CRON' ) && ! WC()->is_rest_api_request();
```

Comment:
This is unrelated to this PR, but I'm surprised there's not a generic WC/WP helper function to tell if it is a frontend request.